src/runtime_src/core/common/query_requests.h

```diff
@@ -1294,24 +1304,82 @@ struct dna_serial_num : request
   }
 };
 
-struct aie_core_info : request
+// Used to retrive aie core tile status information from sysfs
+struct aie_core_info_sysfs : request
+{
+  using result_type = std::string;
+  static const key_type key = key_type::aie_core_info_sysfs;
+
+  virtual boost::any
+  get(const device*) const = 0;
+};
+
+// Used to retrive aie shim tile status information from sysfs
+struct aie_shim_info_sysfs : request
 {
   using result_type = std::string;
-  static const key_type key = key_type::aie_core_info;
+  static const key_type key = key_type::aie_shim_info_sysfs;
 
   virtual boost::any
   get(const device*) const = 0;
 };
 
-struct aie_shim_info : request
+// Used to retrive aie mem tile status information from sysfs
+struct aie_mem_info_sysfs : request
 {
   using result_type = std::string;
-  static const key_type key = key_type::aie_shim_info;
+  static const key_type key = key_type::aie_mem_info_sysfs;
 
   virtual boost::any
   get(const device*) const = 0;
 };
 
+// Retrive aie status version
+// We use binary parser for parsing info from driver
+// This version is used as handshake b/w userspace and driver
+struct aie_status_version : request
+{
+  struct aie_version {
+    uint16_t major;
+    uint16_t minor;
+  };
```

Comment:
yes @dbenusov-xilinx, as this code is common for all aie devices and each device architecture is different we get all these info at runtime.
We are already getting number of columns info
uint32_t cols is for that purpose

Reply:
aie_version structure is for raw buffer we get from device
and the version info in this meta_data structure is needed as there is no common header file b/w user space and driver space, so we get char buffer and reintrepret it. This version info can serve as handshake if in future the structure changes

Reply:
@rbramand-xilinx We discussed that the query struct for aie metadata needed to defined within the query struct. This is required so that we can properly format the buffer for the aie tile data request for windows. You use the tile information to parse the raw data, so we should populate it within this request.